### PR TITLE
chore: improve retry conditions

### PR DESCRIPTION
**Pull request by pyrooka**
This commit adds an extra status code check to the `shouldRetry` method.

### CODE REVIEW

File: src/main/java/com/ibm/cloud/sdk/core/http/RetryInterceptor.java [hunk 1]
```diff
@@ -186,7 +186,7 @@ protected int getInterval(Response response, Request request) {
    */
   protected boolean shouldRetry(Response response, Request request) {
     // First check the response.
-    if (response.code() == 429 || (response.code() >= 500 && response.code() != 501)) {
+    if (response.code() == 429 || (response.code() >= 500 && response.code() <= 599 && response.code() != 501)) {
```
Comment (padamstx) [inline src/main/java/com/ibm/cloud/sdk/core/http/RetryInterceptor.java:189]: I don't think this change has any material effect on things... i don't think we'd expect to ever get a status code > 599 :)
But if we do, then this change nails it :joy:

**Review by padamstx — APPROVED** [review]
LGTM

File: src/main/java/com/ibm/cloud/sdk/core/http/RetryInterceptor.java (hunk 1 above)
Comment (pyrooka) [reply src/main/java/com/ibm/cloud/sdk/core/http/RetryInterceptor.java:189]: I know, but we have the exactly same check in the [Go core](https://github.com/IBM/go-sdk-core/blob/main/v5/core/base_service.go#L789) as well and I just wanted to get it aligned.

**Pull request merged**